### PR TITLE
tulipa

### PR DESCRIPTION
**Pull request by TulipaSilva**

### CODE REVIEW

File: simepar_QC.py [hunk 1]
```diff
@@ -6,23 +6,25 @@ def simepar_write_grid(filename, grid):
...
     
     grid.metadata["origin_latitude"]= grid.axes["lat"]["data"][0]
     grid.metadata["origin_longitude"]= grid.axes["lon"]["data"][0]
     del grid.axes["lat"]
     del grid.axes["alt"]
     del grid.axes["lon"]
-    
+    #del grid.axes['time']['axis']
+    #del grid.axes['time_start']['axis']
+    #del grid.axes['time_end']['axis']
```
Comment (gamaanderson) [inline simepar_QC.py]: this lines shall remain uncommented, all other alteration to this file shall be removed before merge

File: products.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+#field= moment
+
+#grid = pyart.io.read_grid(inputfile, exclude_fields=None) 
+def grid_to_etop_vil_zbase(grid, field):
+
+	if field not in grid.fields.keys():
+		print "ERROR: field not found {} in Netcdf".format(field)
+	matriz = grid.fields[field][ 'data']
```
Comment (gamaanderson) [inline products.py]: This should be:

    raise ValueError("field  %s not found in Grid"%field)

File: products.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	nz = matriz.shape[0]
+	ny = matriz.shape[1]
+	nx = matriz.shape[2]
+	
+	def grid_etop(grid, field, nz, ny, nx):
+		# Carregando as matrizes
```
Comment (gamaanderson) [inline products.py]: don't use function in function, is just ugly!

File: products.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		matriz_alt = np.zeros((nz, ny, nx), dtype='f')
+		altitude = np.zeros((ny, nx), dtype='f')
+		#Tomando as elevações e somando a altura do radar
```
Comment (gamaanderson) [inline products.py]: comment in english!!!

File: products.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		z = np.where(grid.fields[field]["data"]== True)[0].data
+		y = np.where(grid.fields[field]["data"]== True)[1].data
+		x = np.where(grid.fields[field]["data"]== True)[2].data
+		alt = grid.axes["alt"]["data"][0]
```
Comment (gamaanderson) [inline products.py]: try:

		z = np.where(grid.fields[field]["data"].mask == False)[0].data
		y = np.where(grid.fields[field]["data"].mask == False)[1].data
		x = np.where(grid.fields[field]["data"].mask == False)[2].data

File: products.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
+#coding: utf-8
+"""
+Creat three grids. Those containing the fields 'ETOP', 'VILL' and 'Z_base'
```
Comment (gamaanderson) [inline products.py]: Its one grid with tree fields, not tree grids,
By the way this doc should be more like:

    """
    simepar_pyart.products
    ===================
    
    Generate products 'ETOP', 'VILL' and 'ZBASE' from Grid object.
    
    .. autosummary::
        :toctree: generated/
    
        grid_etop
        grid_to_etop_vil_zbase
    """

File: products.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+"""
+
+import copy
+import numpy as np
+import pyart
+import netCDF4
+
+
+def grid_etop(grid, field, nz, ny, nx):
+
```
Comment (gamaanderson) [inline products.py]: Make this return only the field, let to handle axes and metadata to `grid_to_etop_vil_zbase`

File: products.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    """
+    Map a Cartesian grid returning a Grid object whith maximum high cloud.
+
+    Parameters
+    ----------
+    grid: grid from a radar using the function map.grid_from_radars,
```
Comment (gamaanderson) [inline products.py]: grid: Grid object
        Grid object containing volumetric reflectivity data.
    field: string
        Name of the reflectivity field to be used in calculations.

File: products.py [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    in fast_map_to_grid mode.
+    field: One of the fields of grid.fields.keys()
+
+    nz.ny,nx: the grid coordenates
```
Comment (gamaanderson) [inline products.py]: do you need that? just use the shape of the data

File: products.py (hunk 6 above)
Comment (TulipaSilva) [inline products.py]: So, and what about read the grid from a native format from pyart?
It's necessary? How could I do?

Comment (gamaanderson) [inline products.py]: I do not understand the question, grid is already the native structure. Pyart has no real native file format, I have put this point forward (i.e. the need for one) but they didn't agree with me.

File: simepar_QC.py (hunk 1 above)
Comment (gamaanderson) [inline simepar_QC.py]: add a test, if there is axis remove it.

File: products.py [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+    y = tuple(np.where(grid.fields[field]["data"].mask == False)[1])
+    x = tuple(np.where(grid.fields[field]["data"].mask == False)[2])
+    matriz_vil = np.zeros((nz, ny, nx), dtype='f')
+    dh = grid.axes['z_disp']['data'][1] - grid.axes['z_disp']['data'][0]
+    for i in np.arange(len(z)):
+        matriz_vil[z[i], y[i], x[i]] = np.power(np.power( 10.,
+        grid.fields[field]['data'][z[i], y[i], x[i]]/10.), 4./7.)*dh
+    #compute vil and add time axi
+    vil = 3.44e-6*np.sum(matriz_vil, axis=0)[np.newaxis]
+    vil = np.round(vil*128)/128.
```
Comment (gamaanderson) [inline products.py]: You will have to convince me why this better than

    matriz_vil = np.where(grid.fields[field]["data"].mask == False, numpy.power(
                          numpy.power(10., grid.fields[field]["data"] / 10.), 4. / 7.)
                          * dh, 0)
    vil =  3.44e-6 * numpy.sum(matriz_vil, axis=1)

File: products.py [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+
+    #field to grid
+    field_vil = {u'VIL': {u'units': u'kilogram_per_square_meter', u'long_name':
+    u'vertically_integrated_liquid', 'data': np.ma.masked_where(vil == 0, vil,
+    copy=True), u'coordinates': u'elevation azimuth range', u'_FillValue':
+    1e+20}}
+
```
Comment (gamaanderson) [inline products.py]: In this dics I'd ratter have one key per line, but I adjust that while merging

File: products.py [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+    -------
+    grid : Grid
+        A :py:class:`pyart.io.Grid` object containing the gridded radar
+        data.
+        Fields: 'ETOP', 'VILL' and 'Z_base'
+
+
+    """
+    if field not in grid.fields.keys():
+        print "ERROR: field not found {} in Netcdf".format(field)
```
Comment (gamaanderson) [inline products.py]: raise ValueError("field %s not found in Grid" % field)

**Comment by gamaanderson** [comment]
Revise you docs strings and compare to pyart examples, that is all I'm going to say about this.

File: products.py (hunk 10 above)
Comment (TulipaSilva) [inline products.py]: I've never said this.

File: products.py (hunk 11 above)
Comment (TulipaSilva) [inline products.py]: Sorry, what do you mean?

File: products.py (hunk 10 above)
Comment (gamaanderson) [inline products.py]: then I will use the more compact form.

File: products.py (hunk 11 above)
Comment (gamaanderson) [inline products.py]: Not important, I will handle this myself.

**Pull request merged**